x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteForecastAction.java

```diff
@@ -105,24 +106,34 @@ protected void doExecute(Task task, DeleteForecastAction.Request request, Action
             e -> handleFailure(e, request, listener)
         );
 
-        SearchSourceBuilder source = new SearchSourceBuilder();
-
-        BoolQueryBuilder builder = QueryBuilders.boolQuery()
-            .filter(QueryBuilders.termQuery(Result.RESULT_TYPE.getPreferredName(), ForecastRequestStats.RESULT_TYPE_VALUE));
+        BoolQueryBuilder query =
+            QueryBuilders.boolQuery()
+                .filter(QueryBuilders.termQuery(Result.RESULT_TYPE.getPreferredName(), ForecastRequestStats.RESULT_TYPE_VALUE));
         QueryBuilderHelper
             .buildTokenFilterQuery(Forecast.FORECAST_ID.getPreferredName(), forecastIds)
-            .ifPresent(builder::filter);
-        source.query(builder);
-
-        SearchRequest searchRequest = new SearchRequest(AnomalyDetectorsIndex.jobResultsAliasedName(jobId));
-        searchRequest.source(source);
+            .ifPresent(query::filter);
+        SearchSourceBuilder source =
+            new SearchSourceBuilder()
+                .size(MAX_FORECAST_TO_SEARCH)
+                // We only need the following 3 fields, there is no need fetching all of them
+                .fetchSource(
```

Comment:
```suggestion
                .fetchSource(false)
                .docValueField(ForecastRequestStats.FORECAST_ID.getPreferredName())
                .docValueField(ForecastRequestStats. STATUS.getPreferredName())
```

Then the response does not have to be parsed as the fields can be read directly from doc values and the partially populated `ForecastRequestStats` objects can be removed

```
                            for (SearchHit hit : hits) {
                                forecastStatusString = hit.field(ForecastRequestStats.STATUS.getPreferredName()).getValue());
                                if (ignoreStatus(forecastStatusString)) {
                                    continue;
                                }
                                forecastIds.add(hit.field(ForecastRequestStats.FORECAST_ID.getPreferredName()).getValue());
                            }
```

Reply:
Done.

Reply:
thanks